lib/storage/data/file/store.js

```diff
@@ -0,0 +1,274 @@
+'use strict'; // eslint-disable-line
+
+const fs = require('fs');
+const crypto = require('crypto');
+const async = require('async');
+const assert = require('assert');
+
+const Logger = require('werelogs').Logger;
+const errors = require('../../../errors');
+const stringHash = require('../../../stringHash');
+const jsutil = require('../../../jsutil');
+const storageUtils = require('../../utils');
+
+const FOLDER_HASH = 3511;
+
+
+/**
+ * @class
+ * @classdesc File-based object blob store
+ *
+ * Each object/part becomes a file and the files are stored in a
+ * directory hash structure under the configured dataPath.
+ */
+class DataFileStore {
+
+    /**
+     * @constructor
+     * @param {Object} dataConfig - configuration of the file backend
+     * @param {String} dataConfig.dataPath - absolute path where to
+     *   store the files
+     * @param {Boolean} [dataConfig.noSync=false] - If true, disable
+     *   sync calls that ensure files and directories are fully
+     *   written on the physical drive before returning an
+     *   answer. Used to speed up unit tests, may have other uses.
+     * @param {Object} logConfig - logging configuration
+     */
+    constructor(dataConfig, logConfig) {
+        this.logger = new Logger('DataFileStore', logConfig);
+        this.dataPath = dataConfig.dataPath;
+        this.noSync = dataConfig.noSync || false;
+    }
+
+    /**
+     * Setup the storage backend before starting to read or write
+     * files in it.
+     *
+     * The function ensures that dataPath is accessible and
+     * pre-creates the directory hashes under dataPath.
+     *
+     * @param {function} callback - called when done with no argument
+     * @return {undefined}
+     */
+    setup(callback) {
+        fs.accessSync(this.dataPath, fs.F_OK | fs.R_OK | fs.W_OK);
```

Comment:
Now it's more modular it may indeed be a better idea. I just kept how it was done before, since if we can't initialize the backend we can't do much anyway it's not a big deal to assert, but returning a proper async error would be better I agree.